.github/workflows/quality.yml

```diff
@@ -0,0 +1,61 @@
+name: Java Code Quality Analysis
+
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
```

Comment:
The types `[ opened, synchronize, reopened ]` are the default options for the pull_request trigger, so you could omit them here.